cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,238 @@
+	});
+
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		let free_bal =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+		let reserved_bal = <PeopleRococo as PeopleRococoPallet>::Balances::reserved_balance(
+			PeopleRococoSender::get(),
+		);
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		// 3. Set identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			identity_parachain.clone()
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
+
+		// No events get triggered when calling set_sub_no_deposit
+
+		// No amount should be reserved as deposit amounts are set to 0.
+		let reserved_bal = <PeopleRococo as PeopleRococoPallet>::Balances::reserved_balance(
+			PeopleRococoSender::get(),
+		);
+		assert_eq!(reserved_bal, 0);
+		assert!(<PeopleRococo as PeopleRococoPallet>::Identity::identity(
+			&PeopleRococoSender::get()
+		)
+		.is_some());
+		let tuple_subs =
+			<PeopleRococo as PeopleRococoPallet>::Identity::subs_of(&PeopleRococoSender::get());
+		assert!(tuple_subs.1.len() > 0);
+	});
+
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		let free_bal_before_reap =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+		let reserved_balance = <RococoRelay as RococoRelayPallet>::Balances::reserved_balance(
+			RococoRelaySender::get(),
+		);
+		//before reap reserved balance should be equal to total deposit
+		assert_eq!(reserved_balance, total_deposit);
+		assert_ok!(<RococoRelay as RococoRelayPallet>::IdentityMigrator::reap_identity(
+			rococo_runtime::RuntimeOrigin::root(),
+			RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Unreserved { who, amount }) => {
+					who: *who == RococoRelaySender::get(),
+					amount: *amount == total_deposit,
+				},
+			]
+		);
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::identity(&RococoRelaySender::get())
+			.is_none());
+		let tuple_subs =
+			<RococoRelay as RococoRelayPallet>::Identity::subs_of(&RococoRelaySender::get());
+		assert_eq!(tuple_subs.1.len(), 0);
+
+		let reserved_balance = <RococoRelay as RococoRelayPallet>::Balances::reserved_balance(
+			RococoRelaySender::get(),
+		);
+		// after reap reserved balance should be 0
+		assert_eq!(reserved_balance, 0);
+		let free_bal_after_reap =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+
+		// free balance should be greater than before reap
+		assert!(free_bal_after_reap > free_bal_before_reap);
```

Comment:
This should be an `assert_eq` with `free_bal_before_reap + total_deposit`. Asserting > 0 does not fully test the handler

Reply:
Yes. Updated and the assertion is passing.